conversion/conversion.go

```diff
@@ -200,27 +250,82 @@ func snapshotMigrationHandler(sourceProfile profiles.SourceProfile, config write
+		if sourceProfile.Config.ConfigType == "bulk" {
+			var bw *writer.BatchWriter
+
+			//Migrate the data from the data shards, the schema shard needs to be specified here again.
+			for _, dataShard := range sourceProfile.Config.ShardConfigurationBulk.DataShards {
+				//Create a connection profile object for it
+				fmt.Printf("Migrating shard: %v\n", dataShard.DbName)
+				infoSchema, err := getInfoSchemaForShard(dataShard, sourceProfile.Driver, targetProfile)
+				if err != nil {
+					return nil, err
+				}
+				//finally perform a snapshot migration for the shard
+				bw = performSnapshotMigration(config, conv, client, infoSchema)
+			}
+			//finally, once all shard migrations are complete, return the batch writer object
+			return bw, nil
+		} else if sourceProfile.Config.ConfigType == "dataflow" {
+			//STEP - 1 - Create batch for each physical shard
+			asyncProcessShards := func(p *profiles.DataShard, mutex *sync.Mutex) common.TaskResult[*profiles.DataShard] {
+				//create streaming cfg from the config source type.
+				streamingCfg := streaming.CreateStreamingConfig(*p)
+				//update the cfg with the HB defaults
+				err := streaming.VerifyAndUpdateCfg(&streamingCfg, targetProfile.Conn.Sp.Dbname)
+				if err != nil {
+					return common.TaskResult[*profiles.DataShard]{Result: p, Err: err}
+				}
+				//launch the stream for the physical shard
+				err = streaming.LaunchStream(ctx, sourceProfile.Driver, p.LogicalShards, targetProfile.Conn.Sp.Project, streamingCfg.DatastreamCfg)
+				if err != nil {
+					return common.TaskResult[*profiles.DataShard]{Result: p, Err: err}
+				}
+				//perform streaming migration via dataflow
+				err = streaming.StartDataflow(ctx, targetProfile, streamingCfg, conv)
+				return common.TaskResult[*profiles.DataShard]{Result: p, Err: err}
+			}
+			_, err := common.RunParallelTasks(sourceProfile.Config.ShardConfigurationDataflow.DataShards, 5, asyncProcessShards, true)
+			if err != nil {
+				return nil, fmt.Errorf("unable to start minimal downtime migrations: %v", err)
+			}
+			return &writer.BatchWriter{}, nil
+		} else if sourceProfile.Config.ConfigType == "dms" {
+			return nil, fmt.Errorf("dms configType is not implemented yet, please use one of 'bulk' or 'dataflow'")
+		} else {
+			return nil, fmt.Errorf("configType should be one of 'bulk', 'dataflow' or 'dms'")
 		}
-		bw, err := snapshotMigrationHandler(sourceProfile, config, conv, client, infoSchema)
+	default:
+		infoSchema, err := GetInfoSchema(sourceProfile, targetProfile)
 		if err != nil {
 			return nil, err
 		}
-		err = infoSchema.StartStreamingMigration(ctx, client, conv, streamInfo)
-		if err != nil {
-			return nil, err
+		var streamInfo map[string]interface{}
+		if sourceProfile.Conn.Streaming {
+			streamInfo, err = infoSchema.StartChangeDataCapture(ctx, conv)
+			if err != nil {
+				return nil, err
+			}
+			bw, err := snapshotMigrationHandler(sourceProfile, config, conv, client, infoSchema)
```

Comment:
Why is the snapshot hander and streaming handler invoked here ?

Reply:
This is the existing flow of minimal downtime migrations, it is just moved to the `default` switch case.
Ref - 
https://github.com/cloudspannerecosystem/harbourbridge/blob/master/conversion/conversion.go#L214